cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -452,6 +453,31 @@ class alignas(16) column_device_view : public detail::column_device_view_base {
     return string_view{d_strings + offset, d_offsets[index + 1] - offset};
   }
 
+  /**
+   * @brief Returns `device_span` to the list element at the specified index.
+   *
+   * If the element at the specified index is NULL, i.e., `is_null(element_index)
+   * == true`, then any attempt to use the result will lead to undefined behavior.
+   *
+   * This function accounts for the offset.
+   *
+   * @param element_index Position of the desired list element
+   * @return device_span instance representing this element at this index
+   */
+  template <typename T,
+            CUDF_ENABLE_IF(std::is_same_v<T, device_span<int8_t const>> or
+                           std::is_same_v<T, device_span<uint8_t const>>)>
+  [[nodiscard]] __device__ T element(size_type element_index) const noexcept
```

Comment:
I think the element function should be only for column types and not for arbitrary data access for specific use.

Reply:
Maybe create a specific helper function rather than an element method?
```
 template<typename T>
  __device__ get_binary_element(column_device_view const col, size_type element_index)
  {
    using et              = typename T::element_type;
    size_type index       = element_index + col.offset();  // account for this view's _offset
    auto const* d_offsets = col.child(lists_column_view::offsets_column_index)data<offset_type>();
    typename T::element_type const* d_data =
      col.child(lists_column_view::child_column_index).data<et>();
    size_type offset = d_offsets[index];
    return T(d_data + offset, d_offsets[index + 1] - offset);
  }
```
(Preferably not defined in a column header)